lib/health_check.rb

```diff
@@ -0,0 +1,97 @@
+class HealthCheck
+  include ChefResource
+
+  OK = 'ok'.freeze
+  NOT_OK = 'not ok'.freeze
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  AUTHERROR = 'authentication error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = OK
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  #
+  # Shortcut to see if all is well
+  #
+  def ok?
+    @status == OK
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest '_status'
+    end
+  end
+
+  #
+  # Try to authenticate with erchef to prove that oc-id is actually functioning
+  #
+  # TODO: in order for this to work, we need a test user, with
+  # username/password stored in the vault and accessible in Settings
+  #
+  def erchef_authentication
```

Comment:
I'm assuming that you're referring to https://github.com/chef/opscode-platform-cookbooks/pull/742

That check hits `/id/auth/chef/callback` with a GET request, which is part of the OmniAuth middleware we're using, which attempts to login to erchef using no credentials.  Since there are no credentials, erchef refuses to authenticate, OmniAuth detects that failure and returns a 302.  That seems like a great way to check whether or not oc-id can communicate with erchef.

Unfortunately, from the tests that I've done, it doesn't catch the case where oc-id is using an invalid key.  In that case, oc-id will see a 401 from erchef, but the GET to `/id/auth/chef/callback` will still happily redirect like nothing's wrong.

Note that I'm not saying we shouldn't remove this method.  Maybe we should remove it because this isn't something that we want to check.  All I'm saying is that the monitoring that was put in place in the above PR doesn't verify what this method is intended to verify, which is that a valid user can successfully authenticate with erchef, from oc-id.